File: modules/AcroStack.Chat/ContactDto.cs
using System;

namespace AcroStack.Chat;

/// <summary>A user that the current user can start a conversation with.</summary>
public class ContactDto
{
    /// <summary>IdentityUser.Id.</summary>
    public Guid UserId { get; set; }

    public string UserName { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Surname { get; set; } = string.Empty;
    public string? Email { get; set; }

    public int UnreadMessageCount { get; set; }

    public DateTime? LastMessageDate { get; set; }

    /// <summary>Whether the contact is currently online (based on the chat online tracker).</summary>
    public bool IsOnline { get; set; }

    /// <summary>Whether the current user has blocked this contact.</summary>
    public bool IsBlocked { get; set; }
}

File: react/src/api/models/acroStack/chat/ContactDto.ts
/* oxlint-disable */

/**
 * @description A user that the current user can start a conversation with.
 * @type object
 */
export type AcroStackChatContactDto = {
  /**
   * @description IdentityUser.Id.
   *
   * Format: `uuid`
   * @type string | undefined
   */
  userId?: string;
  userName?: string | null;
  name?: string | null;
  surname?: string | null;
  email?: string | null;
  /**
   * @description
   * Format: `int32`
   * @type integer | undefined
   */
  unreadMessageCount?: number;
  /**
   * @description
   * Format: `date-time`
   * @type string | undefined
   */
  lastMessageDate?: string | null;
  /**
   * @description Whether the contact is currently online (based on the chat online tracker).
   * @type boolean | undefined
   */
  isOnline?: boolean;
  /**
   * @description Whether the current user has blocked this contact.
   * @type boolean | undefined
   */
  isBlocked?: boolean;
};
